R/plots_reports.R

```diff
@@ -0,0 +1,176 @@
+#' @rdname plot_OpenSpecy
+#' @title Plots for Open Specy objects
+#'
+#' @description
+#' These functions generate heatmaps, spectral plots, and interactive plots for OpenSpecy data.
+#'
+#' @param object An OpenSpecy object containing metadata and spectral data.
+#' @param z Optional numeric vector specifying the intensity values for the heatmap. If not provided, the function will use the intensity values from the OpenSpecy object.
+#' @param sn Optional numeric value specifying the signal-to-noise ratio threshold. If provided along with \code{min_sn}, regions with SNR below the threshold will be excluded from the heatmap.
+#' @param cor Optional numeric value specifying the correlation threshold. If provided along with \code{min_cor}, regions with correlation below the threshold will be excluded from the heatmap.
+#' @param min_sn Optional numeric value specifying the minimum signal-to-noise ratio for inclusion in the heatmap. Regions with SNR below this threshold will be excluded.
+#' @param min_cor Optional numeric value specifying the minimum correlation for inclusion in the heatmap. Regions with correlation below this threshold will be excluded.
+#' @param selected_spectrum Optional index of the selected spectrum to highlight on the heatmap.
+#' @param x An OpenSpecy object containing metadata and spectral data for the first group.
+#' @param x2 An optional second OpenSpecy object containing metadata and spectral data for the second group.
+#'
+#' @return A plotly heatmap object displaying the OpenSpecy data. A subplot containing the heatmap and spectra plot. A plotly object displaying the spectra from the OpenSpecy object(s).
+#'
+#' @examples
+#' data1 <- read_any(read_extdata("raman_hdpe.json"))
+#' data2 <- read_zip("inst/extdata/CA_tiny_map.zip")
+#' plot_OpenSpecy(data1)
+#' plot_OpenSpecy(data1, x2 = data2)
+#' correlation <- correlate_spectra(conform_spec(data1, new_wavenumbers = data2$wavenumber, res = 5), conform_spec(data2, data2$wavenumbers, res = 5))
+#' heatmap_OpenSpecy(data2, z = data2$metadata$y)
+#' interactive_plot(data1, selected_spectrum = 2)
+#' interactive_plot(data2, selected_spectrum = 2, x2 = data1, selected_spectrum2 = 1)
+#' @author
+#' Win Cowger, Zacharias Steinmetz#'
+#' 
+#' @importFrom plotly plot_ly add_trace add_markers subplot layout
+#' @importFrom tidyr pivot_longer
+#' @importFrom dplyr %>% 
+#' 
+#' @export
+plot_OpenSpecy <- function(x, x2 = NULL, ...) {
+    dt <- cbind(wavenumber = x$wavenumber, x$spectra) |>
+        pivot_longer(cols = -wavenumber, names_to = "id", values_to = "intensity")
```

Comment:
We seem to need the tidyr package only for this `pivot_longer` function. I'll try to look for a replacement to limit the number of package dependencies. Any objections?

Reply:
No objections.